### PR TITLE
[HttpClient] Add `UriTemplateHttpClient`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/17965

### CODE REVIEW

File: http_client.rst [hunk 1]
```diff
@@ -822,6 +823,83 @@ in your requests::
...
+
+    // This request will result on querying http://example.org/users?page=1
+    $client->request('GET', 'http://example.org/{resource}{?page}', [
+        'vars' => [
+            'resource' => 'users',
+            'page' => 1,
+         ],
+    ]);
+
+When using this client in the Framework context, all existing HTTP clients
```
Comment (OskarStark) [inline http_client.rst]:
```suggestion
When using this client in the framework context, all existing HTTP clients
```

**Comment by OskarStark** [comment]
Please check the CI, thanks

**Comment by alexandre-daubois** [comment]
That's updated! 🙂

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
A cool feature and a very nice contribution with good code examples. Thanks Alex!

**Pull request merged**